scaleway/resource_lb.go

```diff
@@ -190,10 +268,72 @@ func resourceScalewayLbUpdate(ctx context.Context, d *schema.ResourceData, meta
 		}
 
 		_, err = lbAPI.UpdateLB(req, scw.WithContext(ctx))
-		if err != nil {
+		if err != nil && !is404Error(err) {
 			return diag.FromErr(err)
 		}
 	}
+	////
+	// Attach / Detach Private Networks
+	////
+	if d.HasChangesExcept("private_network") {
+		retryInterval := DefaultWaitLBRetryInterval
+		// check that pns are in a stable state
+		pns, err := lbAPI.WaitForLBPN(&lb.ZonedAPIWaitForLBPNRequest{
+			Zone:          zone,
+			LBID:          ID,
+			Timeout:       scw.TimeDurationPtr(defaultInstanceServerWaitTimeout),
+			RetryInterval: &retryInterval},
+			scw.WithContext(ctx))
+		if err != nil && !is404Error(err) {
+			return diag.FromErr(err)
+		}
+		// detach private networks
+		for _, pn := range pns {
```

Comment:
If we detach all private networks by default it means that ongoing connections will be lost if there is one between the load-balancer and another host in the private network. We will need to ensure that no connection is dropped before doing a release.